### PR TITLE
Migrate some tests to Spock and eliminate Mockito

### PR DESCRIPTION
**Pull request by jonspalmer**
Spock > Mockito. Also nice to have Groovy in the mix for unit tests.

### CODE REVIEW

File: hibernate-redis/build.gradle [hunk 1]
```diff
@@ -31,8 +32,6 @@ dependencies {
     testCompile group: 'org.hibernate', name: 'hibernate-testing', version:hibernateVersion
     testCompile group: 'junit', name: 'junit', version:'4.11'
     testCompile group: 'org.easytesting', name: 'fest-assert', version:'1.4'
-    testCompile group: 'org.hamcrest', name: 'hamcrest-all', version:'1.3'
-    testCompile group: 'com.carrotsearch', name: 'junit-benchmarks', version:'0.7.2'
```
Comment (jonspalmer) [inline hibernate-redis/build.gradle:35]: can kill a bunch of dependencies.

File: hibernate-redis/src/test/groovy/org/hibernate/test/cache/JedisClientTest.groovy [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+        MultiThreadTestTool.runTasks(100, new Runnable() {
+            @Override
+            public void run() {
+                client.ping()
+                try {
+                    Thread.sleep(1)
+                } catch (InterruptedException ignored) {}
+            }
+        })
+    }
```
Comment (jonspalmer) [inline hibernate-redis/src/test/groovy/org/hibernate/test/cache/JedisClientTest.groovy:58]: I have no idea what this test is doing.

File: hibernate-redis/src/test/groovy/org/hibernate/test/cache/JedisClientTest.groovy [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+        when:
+        long nextTimestamp = client.nextTimestamp("cacheTest")
+
+        then:
+        nextTimestamp == futureTimestamp+1
+    }
+
+    void 'nextTimestamp increments if update fails'() {
+        given:
```
Comment (jonspalmer) [inline hibernate-redis/src/test/groovy/org/hibernate/test/cache/JedisClientTest.groovy:205]: Mocks and Spies are really nice in Spock.

File: hibernate-redis/src/test/groovy/org/hibernate/test/cache/JedisClientTest.groovy (hunk 2 above)
Comment (johnou) [inline hibernate-redis/src/test/groovy/org/hibernate/test/cache/JedisClientTest.groovy:58]: wtf

File: hibernate-redis/build.gradle (hunk 1 above)
Comment (johnou) [inline hibernate-redis/build.gradle:35]: fest-assert?

Comment (jonspalmer) [inline hibernate-redis/build.gradle:35]: used in many of the JUnit tests so can't be killed yet.

**Pull request merged**